numba/typed/dictobject.py

```diff
@@ -912,6 +912,19 @@ def impl(d):
     return impl
 
 
+@overload_method(types.DictType, 'update')
+def ol_dict_update(d, other):
+    if not isinstance(d, types.DictType):
+        return
+    if not isinstance(other, types.DictType):
+        return
```

Comment:
Should `d = {'a':1, 'b':2}; d.update(a=10)` be supported too?

Reply:
I agree this should be supported eventually, but this is meant as a fallback in case the peephole fails and therefore isn't needed for this PR. I think its beyond the scope of this issue and we should open a followup.

Reply:
Done.